### PR TITLE
Fix/package exports

### LINKED ISSUE #770: Invalid exports field in the package.json
This is a **bug report**.

## Problem

We are using [jest](https://jestjs.io/) for running tests on the Status Web. The ecosystem is still in the early adoption of `exports` field, but Jest recently [added](https://github.com/facebook/jest/releases/tag/v28.0.0-alpha.0) supported to the `jest-resolve`.  The tests were failing on the error message `Cannot find module 'js-waku' from <FILE_NAME>`. It took me a while, but I found the culprit. 

#621 introduced invalid `exports` field definition to the `package.json`. The main issue is that [import](https://github.com/status-im/js-waku/blob/master/package.json#L11) condition is not pointing to a valid file that can be loaded via `import` or `import()` (because it is a CommonJS build).

## Proposed Solutions
After reading [Node.js Packages](https://nodejs.org/api/packages.html) documentation and Webpack [Package Exports](https://webpack.js.org/guides/package-exports) article, I believe the correct configuration is the following:

```json
{
  "exports": {
    "node": {
        "import": "./build/esm/index.js",
        "require": "./build/main/index.js",
     }
    "default": "./build/esm/index.js"
  },
}
```

- `import` condition pointing to ESM build
- `require` condition pointing to CommonJS build
- `default` condition is a generic fallback that always matches and should be last

Unless you have environment-specific code, it is not necessary to use the `node` condition. However, it seems to be a good practice to be explicit about it.

### PR DESCRIPTION
**Pull request by prichodko**
Fixes #770.

### CODE REVIEW

**Comment by D4nte** [comment]
Looks good. Can you please add a line in the CHANGELOG @prichodko ?

**Comment by D4nte** [comment]
Will test and merge.

**Comment by D4nte** [comment]
I have published this PR on `@waku/staging@0.24.0-exports.rc0` so it can be tested.

**Comment by D4nte** [comment]
It seems it impacts node application, I need to do more investigation first.

File: package.json [hunk 1]
```diff
@@ -7,10 +7,10 @@
   "module": "build/esm/index.js",
   "exports": {
     "node": {
-      "module": "./build/esm/index.js",
-      "import": "./build/main/index.js"
+      "import": "./build/esm/index.js",
+      "require": "./build/main/index.js"
     },
-    "default": "./build/main/index.js"
+    "default": "./build/esm/index.js"
```
Comment (felicio) [inline package.json:13]: Non-blocking: For consistency, shouldn't `default`'s value be equal to `main`'s even if `exports` takes precedence?

Comment (D4nte) [reply package.json:13]: I think this is the issue I am encountering when using the package in node:

```
/usr/bin/node /home/fryorcraken/.nvm/versions/node/v16.14.0/lib/node_modules/yarn/bin/yarn.js run predefined-test
(node:219199) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)

/home/fryorcraken/src/status-im/waku-tests/node_modules/@waku/staging/build/esm/index.js:1
export { generatePrivateKey, generateSymmetricKey, getPublicKey, } from "./lib/crypto";
^^^^^^

SyntaxError: Unexpected token 'export'
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1032:15)
    at Module._compile (node:internal/modules/cjs/loader:1067:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:168:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:337:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:437:15)
    at async formattedImport (/home/fryorcraken/src/status-im/waku-tests/node_modules/mocha/lib/nodejs/esm-utils.js:7:14)
    at async Object.exports.requireOrImport (/home/fryorcraken/src/status-im/waku-tests/node_modules/mocha/lib/nodejs/esm-utils.js:48:32)
    at async Object.exports.loadFilesAsync (/home/fryorcraken/src/status-im/waku-tests/node_modules/mocha/lib/nodejs/esm-utils.js:103:20)
    at async singleRun (/home/fryorcraken/src/status-im/waku-tests/node_modules/mocha/lib/cli/run-helpers.js:125:3)
    at async Object.exports.handler (/home/fryorcraken/src/status-im/waku-tests/node_modules/mocha/lib/cli/run.js:374:5)

Process finished with exit code 1

```

https://github.com/status-im/waku-tests/compare/master...test-exports-js-waku

**Comment by D4nte** [comment]
new version with the default fix published at `@waku/staging@0.24.0-exports.rc2`

**Comment by D4nte** [comment]
Same error encountered...

**Comment by D4nte** [comment]
I am aiming for the next js-waku version to be ESM only so I won't spend more time on this PR and instead focus on ESM migration which should fix this.

**Comment by D4nte** [comment]
To move forward on this I'd like to confirm js-waku would still work with NodeJS. I used the waku-tests repo as one example but a simpler example is fine too.

**Comment by D4nte** [comment]
@prichodko @felicio please try `@waku/staging@0.24.0-esm.rc8`

See https://github.com/status-im/js-waku/pull/786#issuecomment-1157324872

**Comment by fryorcraken** [comment]
superseded by #803

**Pull request closed without merging**